04_numeronym/pfannerj/main.go

```diff
@@ -0,0 +1,33 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+var mainout io.Writer = os.Stdout
+
+func main() {
+
+	fmt.Fprintln(mainout, numeronyms("accessibility", "Kubernetes", "abc"))
+
+}
+
+func numeronyms(vals ...string) []string {
+
+	valsOut := vals
+	for i := range vals {
+		//Regular expression can be changed or removed altogether if additional chars need to be included
+		regEx := regexp.MustCompile("[^a-zA-Z0-9]+")
+		alphaNumVal := regEx.ReplaceAllString(vals[i], "") //Strips all non-alpha-numeric characters
+		if len(alphaNumVal) > 3 {
+			alphaNumChar := strings.Split(alphaNumVal, "")
+			valsOut[i] = alphaNumChar[0] + strconv.Itoa(len(alphaNumVal)-2) + alphaNumChar[len(alphaNumVal)-1]
```

Comment:
These two lines could be simplified to one as the string doesn't need to be split into chars. Seeing as the first char and the last char are the only values it requires, it could equivalently be completed like so:
```
valOut[i] = alphaNumVal[:1] + strconv.Itoa(len(alphaNumVal)-2) + alphaNumVal[len(alphaNumVal)-1:]
```